moveit_ros/perception/handeye_calibration_target/src/handeye_aruco_target.cpp

```diff
@@ -0,0 +1,301 @@
+    int valid = cv::aruco::estimatePoseBoard(marker_corners, marker_ids, board, camera_matrix_, distortion_coeffs_,
+                                             rvect_, tvect_);
+
+    // Draw the markers and frame axis if at least one marker is detected
+    if (valid == 0)
+    {
+      ROS_DEBUG_STREAM_NAMED(LOGNAME, "Cannot estimate aruco target pose.");
+      return false;
+    }
+
+    if (std::log10(std::fabs(rvect_[0])) > 10 || std::log10(std::fabs(rvect_[0])) > 10 ||
+        std::log10(std::fabs(rvect_[0])) > 10 || std::log10(std::fabs(tvect_[0])) > 10 ||
+        std::log10(std::fabs(tvect_[0])) > 10 || std::log10(std::fabs(tvect_[0])) > 10)
+    {
+      ROS_ERROR_STREAM_NAMED(LOGNAME, "Invalid target pose, please check CameraInfo msg.");
+    }
+
+    cv::Mat image_rgb;
+    cv::cvtColor(image, image_rgb, cv::COLOR_GRAY2RGB);
+    cv::aruco::drawDetectedMarkers(image_rgb, marker_corners);
+    drawAxis(image_rgb, camera_matrix_, distortion_coeffs_, rvect_, tvect_, 0.1);
+    image = image_rgb;
+  }
+  catch (const cv::Exception& e)
+  {
+    ROS_ERROR_STREAM_NAMED(LOGNAME, "Aruco target detection exception: " << e.what());
+    return false;
+  }
+
+  return true;
+}
+
+geometry_msgs::TransformStamped HandEyeArucoTarget::getTransformStamped(std::string& frame_id)
+{
+  geometry_msgs::TransformStamped transform_stamped;
+  transform_stamped.header.stamp = ros::Time::now();
+  transform_stamped.header.frame_id = frame_id;
+  transform_stamped.child_frame_id = "handeye_target";
+
+  tf2::Quaternion q(0, 0, 0, 1);
+  convertToTFQuaternion(rvect_, q);
+  transform_stamped.transform.rotation.x = q.getX();
+  transform_stamped.transform.rotation.y = q.getY();
+  transform_stamped.transform.rotation.z = q.getZ();
+  transform_stamped.transform.rotation.w = q.getW();
+
+  std::vector<double> t(3, 0);
+  convertToStdVector(tvect_, t);
+  transform_stamped.transform.translation.x = t[0];
+  transform_stamped.transform.translation.y = t[1];
+  transform_stamped.transform.translation.z = t[2];
+
+  return transform_stamped;
+}
+
+void HandEyeArucoTarget::convertToTFQuaternion(cv::Vec3d& rvect, tf2::Quaternion& q)
+{
+  cv::Mat rotation_matrix;
+  cv::Rodrigues(rvect, rotation_matrix);
+  if (rotation_matrix.rows == 3 && rotation_matrix.cols == 3)
+  {
+    tf2::Matrix3x3 m;
+    m.setValue(rotation_matrix.ptr<double>(0)[0], rotation_matrix.ptr<double>(0)[1], rotation_matrix.ptr<double>(0)[2],
+               rotation_matrix.ptr<double>(1)[0], rotation_matrix.ptr<double>(1)[1], rotation_matrix.ptr<double>(1)[2],
+               rotation_matrix.ptr<double>(2)[0], rotation_matrix.ptr<double>(2)[1], rotation_matrix.ptr<double>(2)[2]);
+    m.getRotation(q);
+  }
+  else
+  {
+    ROS_ERROR_NAMED(LOGNAME, "Wrong rotation matrix dimensions: %dx%d, it should be 3x3", rotation_matrix.rows,
+                    rotation_matrix.cols);
+  }
+}
+
+void HandEyeArucoTarget::convertToStdVector(cv::Vec3d& tvect, std::vector<double>& t)
```

Comment:
```suggestion
void HandEyeArucoTarget::convertToStdVector(cv::Vec3d& input_vect, std::vector<double>& output_vect)
```

Reply:
Thanks! Changed them and the similar happenings.